src/uint.rs

```diff
@@ -1365,24 +1609,49 @@ where
 {
     type Quotient = PrivateDivQuot<N, D, Q, R, Sub1<UInt<Ui, Bi>>>;
     type Remainder = PrivateDivRem<N, D, Q, R, Sub1<UInt<Ui, Bi>>>;
+
+    fn private_div_if_quotient(self, n: N, d: D, q: Q, r: R, i: UInt<Ui, Bi>, _: Less) -> Self::Quotient
+    where
+    {
+        ().private_div_quotient(n, d, q, r, i - B1)
+    }
+
+    fn private_div_if_remainder(self, n: N, d: D, q: Q, r: R, i: UInt<Ui, Bi>, _: Less) -> Self::Remainder
+    where
+    {
+        ().private_div_remainder(n, d, q, r, i - B1)
+    }
 }
 
 // R == D, I > 0, we set R = 0, Q[I] = 1 and recurse
 impl<N, D, Q, R, Ui, Bi> PrivateDivIf<N, D, Q, R, UInt<Ui, Bi>, Equal> for ()
 where
-    UInt<Ui, Bi>: Sub<B1>,
+    UInt<Ui, Bi>: Copy + Sub<B1>,
```

Comment:
This `Copy` bound is needed because we use `i` both for `q.set_bit(i, B1)` and `i - B1`.